### PR TITLE
fixes QueryResultUtil#getValueByColumnType unsigned int & bigint return value.

### LINKED ISSUE #3231: Issue with data type "INT UNSIGNED"
## Bug Report

1. Create the table
>CREATE TABLE IF NOT EXISTS `table_name` (
  `my_id` int(11) unsigned NOT NULL,
) ENGINE=InnoDB DEFAULT CHARSET=utf8;
2.  I'm trying to extract value: 
> resultSet.getLong("my_id")

Exception occurs: 

> java.sql.SQLDataException: Value '4294967295' is outside of valid range for type java.lang.Integer
	at com.mysql.cj.jdbc.exceptions.SQLError.createSQLException(SQLError.java:114)
	at com.mysql.cj.jdbc.exceptions.SQLError.createSQLException(SQLError.java:97)
	at com.mysql.cj.jdbc.exceptions.SQLError.createSQLException(SQLError.java:89)
	at com.mysql.cj.jdbc.exceptions.SQLError.createSQLException(SQLError.java:63)
	at com.mysql.cj.jdbc.exceptions.SQLError.createSQLException(SQLError.java:73)
	at com.mysql.cj.jdbc.exceptions.SQLExceptionsMapping.translateException(SQLExceptionsMapping.java:92)
	at com.mysql.cj.jdbc.result.ResultSetImpl.getInt(ResultSetImpl.java:805)
	at com.zaxxer.hikari.pool.HikariProxyResultSet.getInt(HikariProxyResultSet.java)
	at org.apache.shardingsphere.core.execute.sql.execute.result.QueryResultUtil.getValueByColumnType(QueryResultUtil.java:59)
	at org.apache.shardingsphere.core.execute.sql.execute.result.QueryResultUtil.getValue(QueryResultUtil.java:45)
	at org.apache.shardingsphere.core.execute.sql.execute.result.StreamQueryResult.getValue(StreamQueryResult.java:75)
	at org.apache.shardingsphere.core.merge.dql.common.StreamMergedResult.getValue(StreamMergedResult.java:56)
	at org.apache.shardingsphere.shardingjdbc.jdbc.core.resultset.ShardingResultSet.getLong(ShardingResultSet.java:112)

Version: Sharding-JDBC 4.0.0RC2

It should extract value as Long.

Mysql jdbc driver in MysqlType.java contains such declaration for it:
> INT_UNSIGNED("INT UNSIGNED", Types.INTEGER, Long.class, MysqlType.FIELD_FLAG_UNSIGNED | MysqlType.FIELD_FLAG_ZEROFILL, MysqlType.IS_DECIMAL, 10L,
            "[(M)] [UNSIGNED] [ZEROFILL]"),

### PR DESCRIPTION
**Pull request by dongzl**
Fixes #3231 .

Changes proposed in this pull request:
- INT UNSIGNED type, if value beyond java Integer range, return Long type.
- INT UNSIGNED type, if value don't beyond java Integer range, return Integer type.
- INT SIGNED type, return Integer type.
- BIGINT UNSIGNED type, if value beyond java Long range, return BigDemical type.
- BIGINT UNSIGNED type, if value don't beyond java Long range, return Long type.
- BIGINT SIGNED type, return Long type.

@tuohai666  Please help me with the code review, Thanks.

### CODE REVIEW

**Comment by dongzl** [comment]
I'm dealing with the the Integration Tests problem, Please wait a moment.

File: sharding-core/sharding-core-execute/src/main/java/org/apache/shardingsphere/core/execute/sql/execute/result/QueryResultUtil.java [hunk 1]
```diff
@@ -105,9 +105,9 @@ private static Object getValueByColumnType(final ResultSet resultSet, final int
             case Types.TINYINT:
             case Types.SMALLINT:
             case Types.INTEGER:
-                return resultSet.getInt(columnIndex);
+                return getIntTypeValue(resultSet, columnIndex);
```
Comment (KomachiSion) [inline sharding-core/sharding-core-execute/src/main/java/org/apache/shardingsphere/core/execute/sql/execute/result/QueryResultUtil.java]: TINYINT and SAMLLINT use getInt is enough

INTEGER if isSigned, should still getInt, if isUnsigned can getLong.

File: sharding-core/sharding-core-execute/src/main/java/org/apache/shardingsphere/core/execute/sql/execute/result/QueryResultUtil.java [hunk 2]
```diff
@@ -135,4 +135,28 @@ private static Object getValueByColumnType(final ResultSet resultSet, final int
                 return resultSet.getObject(columnIndex);
         }
     }
+
+    private static Object getIntTypeValue(final ResultSet resultSet, final int columnIndex) throws SQLException {
```
Comment (KomachiSion) [inline sharding-core/sharding-core-execute/src/main/java/org/apache/shardingsphere/core/execute/sql/execute/result/QueryResultUtil.java]: I think the method can be remove

File: sharding-core/sharding-core-execute/src/main/java/org/apache/shardingsphere/core/execute/sql/execute/result/QueryResultUtil.java [hunk 3]
```diff
@@ -135,4 +135,28 @@ private static Object getValueByColumnType(final ResultSet resultSet, final int
...
+    }
+
+    private static Object getBigIntTypeValue(final ResultSet resultSet, final int columnIndex) throws SQLException {
+        ResultSetMetaData metaData = resultSet.getMetaData();
+        if (metaData.isSigned(columnIndex)) {
+            return resultSet.getLong(columnIndex);
+        }
+        BigDecimal value = resultSet.getBigDecimal(columnIndex);
+        if (new BigDecimal(Long.MAX_VALUE).compareTo(value) <= -1) {
+            return value;
```
Comment (qixiaobo) [inline sharding-core/sharding-core-execute/src/main/java/org/apache/shardingsphere/core/execute/sql/execute/result/QueryResultUtil.java]: maybe return value.toBigInteger()?

File: sharding-core/sharding-core-execute/src/main/java/org/apache/shardingsphere/core/execute/sql/execute/result/QueryResultUtil.java [hunk 4]
```diff
@@ -135,4 +135,28 @@ private static Object getValueByColumnType(final ResultSet resultSet, final int
...
+        }
```
Comment (KomachiSion) [inline sharding-core/sharding-core-execute/src/main/java/org/apache/shardingsphere/core/execute/sql/execute/result/QueryResultUtil.java]: Use BigInteger replace BigDecimal may be better.

can use getBigDecimal().toBigInteger to get BigInteger

File: sharding-core/sharding-core-execute/src/main/java/org/apache/shardingsphere/core/execute/sql/execute/result/QueryResultUtil.java (hunk 3 above)
Comment (dongzl) [reply sharding-core/sharding-core-execute/src/main/java/org/apache/shardingsphere/core/execute/sql/execute/result/QueryResultUtil.java]: > maybe return value.toBigInteger()?

Hi, yes, return `BigInteger` is a good way, as the same with ShardingSphere team review suggestion.

File: sharding-core/sharding-core-execute/src/test/java/org/apache/shardingsphere/core/execute/sql/execute/result/MemoryQueryResultTest.java [hunk 5]
```diff
@@ -56,7 +56,7 @@
 public final class MemoryQueryResultTest {
     
     private final ShardingEncryptor shardingEncryptor = mock(ShardingEncryptor.class);
-
+
```
Comment (terrymanu) [inline sharding-core/sharding-core-execute/src/test/java/org/apache/shardingsphere/core/execute/sql/execute/result/MemoryQueryResultTest.java]: Please keep original indent

File: sharding-core/sharding-core-execute/src/test/java/org/apache/shardingsphere/core/execute/sql/execute/result/MemoryQueryResultTest.java [hunk 6]
```diff
@@ -92,7 +92,7 @@ private EncryptRule getEncryptRule() {
         when(encryptTable.getCipherColumns()).thenReturn(Collections.singleton("order_id"));
         return result;
     }
-
+
```
Comment (terrymanu) [inline sharding-core/sharding-core-execute/src/test/java/org/apache/shardingsphere/core/execute/sql/execute/result/MemoryQueryResultTest.java]: Please keep original indent

**Pull request merged**